lib/hooks/useDateFormatter.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from 'react';
+import useSiteData from './useSiteData';
+
+/**
+ * Gets forecasted and solar panel data for a single site
+ * @param long the longitude value of the user's site
+ * @param lat the latitude value of the user's site
+ * @returns the users timezone based on mapbox's API, this returns GMT otherwise
+ */
+
+const fetchTimeZone = async (long: number, lat: number) => {
+  const query = await fetch(
+    `https://api.mapbox.com/v4/examples.4ze9z6tv/tilequery/${long},${lat}.json?access_token=${process.env.NEXT_PUBLIC_MAPBOX_PUBLIC}`,
+    { method: 'GET' }
+  );
+  const data = await query.json();
+  // get the timezone from the resulting GeoJSON FeatureCollection
+  const userTimezone = data?.features[0]?.properties?.TZID ?? 'GB';
```

Comment:
perhaps leave a comment here describing behavior when the time zone is not found—the assumption of `'GB'` might not be valid in the very long term, so we want to make clear that this is an assumption here.